### DISCUSSION TITLE
How to make one task depends on the previous day result of another task?

### DISCUSSION THREAD

**Discussion by qianyun210603** [apache/airflow#32651] [General]
I have a task A which reads dataset A on day T-1; another task B which reads dataset B on day T; and finally task C to calculate results using dataset A and B. So I hope to setup some DAG which can let the run of task C on day T depends on the status of task A on day T-1 and task B on day T. How can I do this elegantly？

So the really question is how to make task C's day T run depends on task A's day T-1 run. I digged a bit with help of ChatGPT, it provides solution using `ExternalTaskSensor`, something like:
```python
task_helper = ExternalTaskSensor(
    task_id='wait_for_prev_day_task_A',
    external_dag_id="thisdag",
    external_task_id='task_A',
    execution_delta=timedelta(days=1),  # Same schedule interval as the DAG
    check_existence=True,
    failed_states=[State.FAILED, State.SKIPPED, State.SHUTDOWN, State.UPSTREAM_FAILED],
    dag=thisdag,
)

task_C = BashOperator(...)

task_helper >> task_C
```

But the problem is in the first day execution (the `start_date` of the DAG), the task_helper keeps searching the non-existing task_A (since the DAG hasn't start), even I set `check_existence` to true.

How should I avoid this?

**Comment by potiuk** [comment]
You can't do it this way. DAGs are scheduled as a "whole" on the same schedule. You can likely do it by having separate DAGs and triggering things around but your use case is pretty niche.
